### PR TITLE
Optimize ixalloc by avoiding a size lookup.

### PR DESCRIPTION
**Pull request by interwq**

### CODE REVIEW

**Comment by interwq** [comment]
This avoids an unnecessary `isalloc(tsdn, ptr)` on the `xallocx` path.

File: src/arena.c [hunk 1]
```diff
@@ -1606,17 +1608,23 @@ arena_ralloc_no_move(tsdn_t *tsdn, void *ptr, size_t oldsize, size_t size,
...
-		return false;
+		ret = false;
 	} else if (oldsize >= LARGE_MINCLASS && usize_max >= LARGE_MINCLASS) {
-		return large_ralloc_no_move(tsdn, extent, usize_min, usize_max,
+		ret = large_ralloc_no_move(tsdn, extent, usize_min, usize_max,
 		    zero);
+	} else {
+		ret = true;
 	}
+lable_done:
```
Comment (djwatson) [inline src/arena.c]: spelling - label_done? Or even better, just done:

File: src/arena.c [hunk 2]
```diff
@@ -1644,7 +1652,9 @@ arena_ralloc(tsdn_t *tsdn, arena_t *arena, void *ptr, size_t oldsize,
 
 	if (likely(usize <= SMALL_MAXCLASS)) {
 		/* Try to avoid moving the allocation. */
-		if (!arena_ralloc_no_move(tsdn, ptr, oldsize, usize, 0, zero)) {
+		size_t newsize;
```
Comment (djwatson) [inline src/arena.c]: I like to name these types of things 'unused' for clarity

File: src/arena.c (hunk 1 above)
Comment (interwq) [reply src/arena.c]: Ah good catch. Will fix.

File: src/arena.c (hunk 2 above)
Comment (interwq) [reply src/arena.c]: Agreed. Will add.

**Comment by interwq** [comment]
Addressed feedback @djwatson.

**Pull request merged**